# Sjogren syndrome microarray data analysis

Topics to be covered:

* Data preparation, outliers identification and their effect on functional enrichment analyses
* Using ComBat to account for batch effect
* KEGG, Reactome, GO enrichment analyses
* Genes best correlated with clinical parameters, ant their functional significance
* TODO

```{r setup, echo=FALSE, include=FALSE, cache=FALSE}
# Set up the environment
library(knitr) 
opts_chunk$set(cache.path='cache/', fig.path='img/', cache=F, tidy=T, fig.keep='high', echo=F, dpi=300, out.width=700, message = FALSE, warning = FALSE)
options(replace.assign=TRUE, width=120)
set.seed(1)
HIDE=F # If true, mask any data labels
```

```{r loadLibraries, echo=FALSE, warning=FALSE, message=FALSE}
# Load necessary packages
library(MDmisc)
library(ReactomePA)
library(annotables)
library(xlsx)
library(dplyr)
suppressMessages(library(org.Hs.eg.db))
suppressMessages(library(limma))
suppressMessages(library(genefilter))
suppressMessages(library(sva))
suppressMessages(library(gplots))
suppressMessages(library(gridExtra))
suppressMessages(library(Hmisc))
```

```{r clinCorrel, echo=FALSE}
# A wrapper function to visualize top 20 genes best correlated with a clinical parameter,
# and perform pathway/GO enrichment analysis on top 100 best correlated genes
clinCorrel <- function(m, fileName=NULL, edata=edata) {
  corrgenes <- apply(edata, 1, function(x) rcorr(x, m))
  corrgenes <- cbind(sapply(corrgenes, function(x) x$r[1, 2]), 
               sapply(corrgenes, function(x) x$P[1, 2]))
  colnames(corrgenes) <- c("CorrCoeff", "Pval")
  corrgenes <- merge(corrgenes, annot.f, by="row.names")
  corrgenes <- corrgenes[order(corrgenes[, "CorrCoeff"], decreasing=T), ]
  rownames(corrgenes) <- corrgenes$Row.names
  
  if (!is.null(fileName)) {
    write.table(corrgenes[,c(2,3,5,7)], paste("results//", fileName, sep=""), sep="\t", row.names=F)
  }
#  if (nrow(corrgenes) > 20) {t <- 20} else {t <- nrows(corrgenes)}
#  grid.table(corrgenes[1:t, c(2,3,5,7)], gp=gpar(fontsize=6), core.just="left")
  return(corrgenes)
}
```

```{r gene_enrichment, echo=FALSE}
# A wrapper function to perform all functional enrichment analyses. The 'fileName' should have 'xlsx' extension
all.enrichment.analyses <- function(genes, fileName, id="symbol") {
  # Gene ontology, molecular function
  res <- gene_enrichment(selected = genes, id=id, use="GO", ont="MF")
  write.xlsx(res, fileName, sheetName = "GO-MF", row.names=FALSE, append=TRUE)
  # Gene ontology, biological process 
  res <- gene_enrichment(selected = genes, id=id, use="GO", ont="BP")
  write.xlsx(res, fileName, sheetName = "GO-BP", row.names=FALSE, append=TRUE)
  # Gene ontology, cellular component
  res <- gene_enrichment(selected = genes, id=id, use="GO", ont="CC")
  write.xlsx(res, fileName, sheetName = "GO-CC", row.names=FALSE, append=TRUE)
  # KEGG canonical pathways
  res <- gene_enrichment(selected = genes, id=id, use="KEGG")
  write.xlsx(res, fileName, sheetName = "KEGG", row.names=FALSE, append=TRUE)
}
```

```{r loadDataFunctions, echo=FALSE}
# Load meta- and expression data, specifying outlier IDs
loadMeta <- function(outliersRemove)
{ # Load meta data
  p <- patients
  if (outliersRemove) {p <- patients[!patients %in% c("p1033216.2", "p1033680.6...5.")]}
  meta <- as.data.frame(t(meta12[, p])) # Subsetting
  colnames(meta) <- sapply(colnames(meta), function(x) gsub(" ", "", x)) # Removing spaces from column names
  write.table(meta, "data//meta.txt", sep="\t", row.names=T) # Store loaded meta data
  return(meta)
}

loadExprs <- function(outliersRemove)
{ # Load expression data
  p <- patients
  if (outliersRemove) {p <- patients[!patients %in% c("p1033216.2", "p1033680.6...5.")]}
  exprs.n <- log2(normalizeQuantiles(exprs[, p]))
  colnames(exprs.n) <- colnames(exprs[, p])
  rownames(exprs.n) <- paste("gene", seq(1,nrow(exprs.n)), sep="") # Dummy row names, to be later used for merging
  return(exprs.n) # Subsetting
}

```

```{r prinComponents, echo=FALSE}
prinComponents <- function(exprs, meta)
{
  #summary(prcomp(log10(exprs)))
  pca<-prcomp(exprs)$rotation
  x = pca[,1]; y = pca[,2]
  xadj<-0.1*(max(x)-min(x)); yadj<-0.1*(max(y)-min(y)) 
  plot(x,y,xlab="PC1", ylab="PC2", main="PC analysis on cohorts",
     xlim=c(min(x) - xadj, max(x) + xadj), ylim=c(min(y) - yadj, max(y) + yadj),
     pch=ifelse(meta$Cohort == 1, 1, 2),
     col=ifelse(meta$MicroarrayClass == "AC", "red", "blue")) 
  if (!HIDE) text(x,y+0.03,labels=rownames(pca), cex=0.7)
  legend("bottomright", c("Cohort1/AC", "Cohort2/AC", "Cohort1/PCC", "Cohort2/PCC"), 
      col=c("red","red","blue","blue"),
      pch=c(1,2,1,2))
}
```

```{r limma, echo=FALSE}
# Limma on the expression matrix and a level from meta
limmaOnData <- function(exprs, meta, modFactor, fileName=NULL)
{
  mod<-model.matrix(~as.factor(eval(parse(text=modFactor))), data=meta) # The meta data should exist
  colnames(mod)[2]<-"outcome"
  eset.0 <- new("ExpressionSet", exprs=as.matrix(exprs))
  fit <- lmFit(eset.0, mod)
  fit2 <- eBayes(fit)
  res <- topTable(fit2, coef="outcome", number=nrow(exprs(eset.0)), adjust.method="BH", p.value=0.1)
  if (!is.null(fileName)) {
    write.table(merge(res, annot.f, by="row.names"), paste("results//", fileName, sep=""), sep="\t", row.names=F)
  }
  return(res)
}
```

## Data preparation

`annot.txt` is taken from `Single-experiment raw data3//annot3.txt`

`data.txt` are combined from `01.Data.Analysis.xlsx` and `11.Data.xlsx`.

`meta1.txt` is taken from `11 Annotations final cohorts sheet 2 20DEC13 DF.xlsx` and contains cohort information. The data has been transposed  to have patients names as columns for compatibility with `data.txt`

`meta2.txt` is taken from `20140214FarrisMicroarraymgdb(1).xlsx` (accessed on 03-13-2014). The data has been transposed. ID `p1033680-6` has been renamed to `p1033680-6 (-5)` to be compatible with the `data.txt` header.

```{r loadData, echo=FALSE}
# Full annotations. To avoid R errors on special characters, in Excel select "Description" column, use Ctrl+1, Custom category, \"@\" type
annot.f <- as.data.frame(read.table("data//annot_full.txt",sep="\t", quote="\"", header=T, as.is=T))
annot.f <- apply(annot.f, 2, function(x){sapply(x, function(y){gsub("\"", "", y)})}) # Remove \"
colnames(annot.f) <- c("ProbeName", "GeneName", "SystematicName", "Description") # Rename columns
rownames(annot.f) <- paste("gene", seq(1, nrow(annot.f)), sep="") # Dummy row names, to be later used for merging
# Expression data
exprs <- as.matrix(read.table("data//data.txt", sep="\t", header=T))
# Meta data
meta1 <- read.table("data//meta1.txt", sep="\t", header=T, row.names=1, as.is=T) # Has cohort
meta2 <- read.table("data//meta2.txt", sep="\t", header=T, row.names=1, as.is=T) # Has other clinical parameters
meta12 <- as.data.frame(rbind(meta1[1:2, intersect(colnames(meta1), colnames(meta2))], 
                              meta2[, intersect(colnames(meta1), colnames(meta2))])) # Need only the first two parameters from meta1. Join vertically by common names
patients <- colnames(meta12)[meta12["Microarray Class", ] == "AC" | meta12["Microarray Class", ] == "PSS"] # We use only two treatment conditions. Should be 34 patients. 
```

To identify the largest source of variability within the data, we perform principal component analysis and color the samples by Cohort/Treatment status.

```{r dataWithOutliers}
outliersRemove <- FALSE
meta.withOutliers <- loadMeta(outliersRemove)
exprs.n.withOutliers <- loadExprs(outliersRemove)
# arrayQualityMetrics(new("ExpressionSet", exprs=exprs.n), outdir="arrayQC_WithOutliers")
```

```{r PCAwithOutliers}
prinComponents(log2(exprs.n.withOutliers), meta.withOutliers)
```

Clearly, the cohorts are very different. The two patients, p1033216.2 and p1033680.6...5., appear as outliers. They are also picked up by the _arrayQualityMetrics_ set of tests. 

We remove them and look at the principal components again.

```{r dataWithoutOutliers}
outliersRemove <- TRUE
meta <- loadMeta(outliersRemove)
exprs.n <- loadExprs(outliersRemove)
# arrayQualityMetrics(new("ExpressionSet", exprs=exprs.n), outdir="arrayQC_WithoutOutliers")
```

```{r PCAwithoutOutliers}
prinComponents(log2(exprs.n), meta)
```

The data looks more homogeneous now, with the cohort effect still dominating.

To further investigate how the outliers affect differential gene expression, we will perform differential expression _limma_ analysis on the expression data with and without outlier. 

### Differential expression with outliers

```{r limmaWithOutliers}
res <- limmaOnData(exprs.n.withOutliers, meta.withOutliers, "MicroarrayClass")
# res.pathway <- enrichPathway(gene = grch37$entrez[ grch37$symbol %in% annot.f[rownames(res), "GeneName"] %>% as.vector ],
#                              organism = "human", pAdjustMethod = "BH", pvalueCutoff = 0.1, qvalueCutoff = 0.2, readable = T)
# if (nrow(summary(res.pathway)) > 0) {
#   summary(res.pathway) %>% kable
# }
```

Only `r nrow(res)` probes are differentially expressed out of `r nrow(exprs.n.withOutliers)` total. 

### Differential expression without outliers

```{r limmaWitouthOutliers, dependson="dataWithoutOutliers"}
res <- limmaOnData(exprs.n, meta, "MicroarrayClass")
# res.pathway <- enrichPathway(gene = grch37$entrez[ grch37$symbol %in% annot.f[rownames(res), "GeneName"] %>% as.vector ],
#                              organism = "human", pAdjustMethod = "BH", pvalueCutoff = 0.1, qvalueCutoff = 0.2, readable = T)
# if (nrow(summary(res.pathway)) > 0) {
#   summary(res.pathway) %>% kable
# }
```

Removing the outliers improved sensitivity in detecting differentially expressed genes, `r nrow(res)` DEGs.

## Using ComBat to account for batch effect

To investigate batch effect in raw data, we will test the differences between the cohorts of patients, without considering their treatment status. 

The outliers were removed.

```{r limmaCohorts, dependson="dataWithoutOutliers"}
res <- limmaOnData(exprs.n, meta, "Cohort")
```

Almost 1/3 of the probes are differentially expressed between cohorts, `r nrow(res)` DEGs out of `r nrow(exprs.n)` total.

We use ComBat to manually adjust for the cohort effect and keep the treatment effect. If we look at the PCA plot after the adjustment, the data look more homogeneous, and the cohorts are now mixed together.

```{r combat, dependson="dataWithoutOutliers"}
combat_edata <- ComBat(dat=exprs.n, batch=meta$Cohort, mod=model.matrix(~as.factor(MicroarrayClass), data=meta), par.prior=TRUE, prior.plots=F)
prinComponents(combat_edata, meta)
```

```{r limmaOnCohortCombat, dependson="combat"}
res <- limmaOnData(combat_edata, meta, "Cohort")
```

And no differentially expressed genes between the cohorts can be detected after removing the batch effect, `r nrow(res)` DEGs.

# Differential expression analysis

So far we used the complete dataset with 62976 probes. However, probes constantly expressed across the conditions are less interesting to us. To increase power of detecting differentially expressed genes, we remove such constantly expressed genes.

```{r dataFilter, dependson="combat"}
# Filtering low variability probes
combat_edata <- exprs(varFilter(new("ExpressionSet", exprs=combat_edata)))
```

Almost half of the probes were removed, keeping `r nrow(combat_edata)` probes. 

```{r limmaOnTreatmentCombat, dependson="combat"}
res <- limmaOnData(combat_edata, meta, "MicroarrayClass", "limma_MicroarrayClass_WithoutOutliers_WithoutBatch.txt")
```

After removing batch effect, we get more differentially expressed genes, `r nrow(res)` DEGs.

Let's have a look at top 50 differentially expressed genes. Multiple probes for the same gene are collapsed to one by maximum expression level.

```{r limmaVisual, warning=FALSE, fig.height=9, dependson="combat"}
degs <- WGCNA::collapseRows(combat_edata[rownames(res), ], annot.f[rownames(res), "GeneName"], rownames(res))
# Sort the collapsed data by the largest absolute fold change
maxRatio <- rowMeans(degs$datETcollapsed[, meta$MicroarrayClass == "PSS"])-
           rowMeans(degs$datETcollapsed[, meta$MicroarrayClass == "AC"])
degs.sorted <- degs$datETcollapsed[order(abs(maxRatio), decreasing=T), 
                                  order(meta$MicroarrayClass)]
# Or just use the most significant data, columns reordered
# degs.sorted <- degs$datETcollapsed[, order(meta$MicroarrayClass)]
color <- colorRampPalette(c("blue", "white", "red"))
if (HIDE) labCol="" else labCol=colnames(degs.sorted)
if (HIDE) labRow="" else labRow=rownames(degs.sorted)
heatmap.2(degs.sorted[1:50, ], Colv=F, Rowv=F, scale="row", trace="none", col=color, key=T, density.info="none", cexCol=1, cexRow=0.8, labCol=labCol, labRow=labRow[1:50])#, cellnote=formatC(1/10^abs(mtx), format="e", digits=2), notecol='darkgreen')
```

```{r aggregateDegs}
# Merge limma results and annotations
tmp <- merge(res, annot.f, by="row.names")
# Get unique "gene name - max logFC" pairs"
degs <- aggregate(tmp[, "logFC"], by=list(tmp$GeneName), max)
degs.tmp <- merge(degs, tmp, by.x="x", by.y="logFC")
# Save the results
fileName <- "results/Cohort_DEGs_Enrichment.xlsx"
unlink(fileName)
write.xlsx(degs.tmp, fileName, sheetName = "DEGs", row.names = F, append = T)
```

We aggregate probe names summarizing expression of the same gene by maximum fold change. The list of all differentially expressed genes, and the full statistical output, are in the "`r fileName`" file.

The results of functional enrichment analysis of these differentially expressed genes are in the same "`r fileName`" file. 

Legend: "GO" - gene ontology, with focus on "MF"/"BP"/"CC" - molecular functions/biological processes/cellular components functions; "KEGG" - canonical pathways.

Note: For any KEGG pathway, we can visualize it as a figure, and overlay fold changes of differentially expressed genes

```{r dependson="limmaOnTreatmentCombat"}
all.enrichment.analyses( annot.f[ rownames(res), "GeneName"] %>% as.vector %>% unique, fileName)
# # Reactome analysis
# gene = grch37$entrez[ grch37$symbol %in% annot.f[rownames(res), "GeneName"] %>% as.vector]
# gene = gene[ !is.na(gene)]
# res.pathway <- enrichPathway(gene, organism = "human", pAdjustMethod = "BH", pvalueCutoff = 0.1, qvalueCutoff = 0.2, readable = T)
# if (nrow(summary(res.pathway)) > 0) {
#   summary(res.pathway) %>% kable
# }
```

# Genes best correlating with clinical parameters

Interesting clinical parameters are: WUSFvol, SSFvolL, SSFvolR, LGleft, LGright, FS, LaBioRadvalue, RoBioRadvalue, RFvalue. For each clinical parameter, we answer 3 questions:

1) What are the top 20 genes best correlated with a clinical parameter?
2) Are those genes enriched in canonical pathways? Which? In how many?
3) Are those genes enriched in "biological process" gene ontologies? Which? In how many?

We can look at correlations in multiple ways:

1) Clinical parameter with all genes, e.g., clinCorrel(meta$WUSFvol, , combat_edata)
2) Clinical parameter with DEGs only, e.g., clinCorrel(meta$WUSFvol, , combat_edata[rownames(res), ])
3) Current. Clinical parameter with DEGs only in PSS group, e.g., 

Note, GO and KEGG enrichment analyses are done using genes correlating with clinical parameters with p-value < 0.05.

WUSFvol

```{r WUSFvol_c1, fig.width=12}
fileName <- "results/clinCorrel_WUSFvol.xlsx"
unlink(fileName)
correlGenes <- clinCorrel(meta$WUSFvol[meta$MicroarrayClass == "PSS"], , combat_edata[rownames(res), names(meta$FS[meta$MicroarrayClass == "PSS"])])
write.xlsx2(correlGenes, fileName, sheetName = "correlGenes", row.names = F, append = T)
all.enrichment.analyses(correlGenes$GeneName %>% unique, fileName)
```

SSFvolL

```{r SSFvolL_c1, fig.width=12}
fileName <- "results/clinCorrel_SSFvolL.xlsx"
unlink(fileName)
correlGenes <- clinCorrel(meta$SSFvolL[meta$MicroarrayClass == "PSS"], , combat_edata[rownames(res), names(meta$FS[meta$MicroarrayClass == "PSS"])])
write.xlsx2(correlGenes, fileName, sheetName = "correlGenes", row.names = F, append = T)
all.enrichment.analyses(correlGenes$GeneName %>% unique, fileName)
```

SSFvolR

```{r SSFvolR_c1, fig.width=12}
fileName <- "results/clinCorrel_SSFvolR.xlsx"
unlink(fileName)
correlGenes <- clinCorrel(meta$SSFvolR[meta$MicroarrayClass == "PSS"], , combat_edata[rownames(res), names(meta$FS[meta$MicroarrayClass == "PSS"])])
write.xlsx2(correlGenes, fileName, sheetName = "correlGenes", row.names = F, append = T)
all.enrichment.analyses(correlGenes$GeneName %>% unique, fileName)
```

LGleft

```{r LGleft_c1, fig.width=12}
fileName <- "results/clinCorrel_LGleft.xlsx"
unlink(fileName)
correlGenes <- clinCorrel(meta$LGleft[meta$MicroarrayClass == "PSS"], "clinCorrel.LGleft.genes", combat_edata[rownames(res), names(meta$FS[meta$MicroarrayClass == "PSS"])])
write.xlsx2(correlGenes, fileName, sheetName = "correlGenes", row.names = F, append = T)
all.enrichment.analyses(correlGenes$GeneName %>% unique, fileName)
```

LGright

```{r LGright_c1, fig.width=12}
fileName <- "results/clinCorrel_LGright.xlsx"
unlink(fileName)
correlGenes <- clinCorrel(meta$LGright[meta$MicroarrayClass == "PSS"], , combat_edata[rownames(res), names(meta$FS[meta$MicroarrayClass == "PSS"])])
write.xlsx2(correlGenes, fileName, sheetName = "correlGenes", row.names = F, append = T)
all.enrichment.analyses(correlGenes$GeneName %>% unique, fileName)
```

FS

```{r FS_c1, fig.width=12}
fileName <- "results/clinCorrel_FS.xlsx"
unlink(fileName)
correlGenes <- clinCorrel(meta$FS[meta$MicroarrayClass == "PSS"], , combat_edata[rownames(res), names(meta$FS[meta$MicroarrayClass == "PSS"])])
write.xlsx2(correlGenes, fileName, sheetName = "correlGenes", row.names = F, append = T)
all.enrichment.analyses(correlGenes$GeneName %>% unique, fileName)
```

LaBioRadvalue

```{r LaBioRadvalue_c1, fig.width=12}
fileName <- "results/clinCorrel_LaBioRadvalue.xlsx"
unlink(fileName)
correlGenes <- clinCorrel(meta$LaBioRadvalue[meta$MicroarrayClass == "PSS"], , combat_edata[rownames(res), names(meta$FS[meta$MicroarrayClass == "PSS"])])
write.xlsx2(correlGenes, fileName, sheetName = "correlGenes", row.names = F, append = T)
all.enrichment.analyses(correlGenes$GeneName %>% unique, fileName)
```

RoBioRadvalue

```{r RoBioRadvalue_c1, fig.width=12}
fileName <- "results/clinCorrel_RoBioRadvalue.xlsx"
unlink(fileName)
correlGenes <- clinCorrel(meta$RoBioRadvalue[meta$MicroarrayClass == "PSS"], , combat_edata[rownames(res), names(meta$FS[meta$MicroarrayClass == "PSS"])])
write.xlsx2(correlGenes, fileName, sheetName = "correlGenes", row.names = F, append = T)
all.enrichment.analyses(correlGenes$GeneName %>% unique, fileName)
```

RFvalue

```{r RFvalue_c1, fig.width=12}
fileName <- "results/clinCorrel_RFvalue.xlsx"
unlink(fileName)
correlGenes <- clinCorrel(meta$RFvalue[meta$MicroarrayClass == "PSS"], , combat_edata[rownames(res), names(meta$FS[meta$MicroarrayClass == "PSS"])])
write.xlsx2(correlGenes, fileName, sheetName = "correlGenes", row.names = F, append = T)
all.enrichment.analyses(correlGenes$GeneName %>% unique, fileName)
```

# Genes best correlating with the flow counts

```{r loadFlowData, echo=FALSE}
flow <- read.table("data//flow.txt", sep="\t", header=T, row.names=1, stringsAsFactors = F)
rownames(flow) <- make.names(rownames(flow)) # Make row names compatible with the combat_edata
flow.mtx <- as.matrix(flow[1:(nrow(flow) - 1), 4:ncol(flow)]) # Remove last patient
flow.patients <- intersect(rownames(flow.mtx), colnames(combat_edata))
```

We first check how the flow measurements correlate with each other. Numbers in each cell indicate Pearson's correlation coefficient. It is expected that a measurement will correlate with itself with the coefficient equal to "1".

Things to watch for: Some different measurements show perfect correlation of "1". Should it be so?

```{r flowCorrel}
selected <- flow.mtx[,1:11] # Flow counts as percentage
selected <- flow.mtx[,13:20] # Flow counts as numbers
class(selected) <- "numeric"
flow.cor <- rcorr(selected[complete.cases(selected), ]) 
# pdf(paste("results", "flowCorrel.pdf", sep="/"))
par(oma=c(5,0,0,5), mar=c(6, 4.1, 4.1, 6)) # Adjust margins
color<-colorRampPalette(c("blue","yellow")) # Define color gradient
# Adjust clustering parameters.
# Distance: "euclidean", "maximum","manhattan" or "minkowski". Do not use "canberra" or "binary"
# Clustering: "ward", "single", "complete", "average", "mcquitty", "median" or "centroid"
dist.method<-"euclidean"  
hclust.method<-"ward.D"
# Make Pearson's coefficients and p-values merged in one cell
cnote <- matrix(0, nrow=nrow(flow.cor$r), ncol=ncol(flow.cor$r))
for (i in 1:nrow(flow.cor$r)) {
  for (j in 1:ncol(flow.cor$r)) {
    cnote[i, j] <- paste(formatC(flow.cor$r[i, j], format="f", digits=2), formatC(flow.cor$P[i, j], format="e", digits=2), sep="\n")
  }
}
h<-heatmap.2(as.matrix(flow.cor$r), trace="none", density.info="none", col=color,distfun=function(x){dist(x, method=dist.method)}, hclustfun=function(x){hclust(x, method=hclust.method)}, cexRow=0.7, cexCol=0.7, cellnote=cnote , notecex=0.6, notecol='red')
# dev.off()
```

%CD4+RA-(of CD3+)

```{r pctCD4plusRAminus.of.CD3plus._c1, fig.width=12}
fileName <- "results/flowCorrel_pctCD4plusRAminus.of.CD3plus..genes.xlsx"
unlink(fileName)
correlGenes <- clinCorrel(flow.mtx[, "pctCD4plusRAminus.of.CD3plus."], , combat_edata[rownames(res), flow.patients])
write.xlsx2(correlGenes, fileName, sheetName = "correlGenes", row.names = F, append = T)
all.enrichment.analyses(correlGenes$GeneName %>% unique, fileName)
```

%CD8+RA-(of CD3+)    

```{r pctCD8plusRAminus.of.CD3plus._c1, fig.width=12}
fileName <- "results/flowCorrel_pctCD8plusRAminus.of.CD3plus..genes.xlsx"
unlink(fileName)
correlGenes <- clinCorrel(flow.mtx[, "pctCD8plusRAminus.of.CD3plus."], , combat_edata[rownames(res), flow.patients])
write.xlsx2(correlGenes, fileName, sheetName = "correlGenes", row.names = F, append = T)
all.enrichment.analyses(correlGenes$GeneName %>% unique, fileName)
```

%CD4-8-(of CD3+)

```{r pctCD4minus8minus.of.CD3plus._c1, fig.width=12}
fileName <- "results/flowCorrel_pctCD4minus8minus.of.CD3plus..genes.xlsx"
unlink(fileName)
correlGenes <- clinCorrel(flow.mtx[, "pctCD4minus8minus.of.CD3plus."], , combat_edata[rownames(res), flow.patients])
write.xlsx2(correlGenes, fileName, sheetName = "correlGenes", row.names = F, append = T)
all.enrichment.analyses(correlGenes$GeneName %>% unique, fileName)
```

ratio.of.pctCD4plusRAminusperpct.CD8plusRAminus  

```{r ratio.of.pctCD4plusRAminusperpct.CD8plusRAminus_c1, fig.width=12}
fileName <- "results/flowCorrel_ratio.of.pctCD4plusRAminusperpct.CD8plusRAminus.genes.xlsx"
unlink(fileName)
correlGenes <- clinCorrel(flow.mtx[, "ratio.of.pctCD4plusRAminusperpct.CD8plusRAminus"], , combat_edata[rownames(res), flow.patients])
write.xlsx2(correlGenes, fileName, sheetName = "correlGenes", row.names = F, append = T)
all.enrichment.analyses(correlGenes$GeneName %>% unique, fileName)
```

pctCD38hi27hi.of.TN.

```{r pctCD38hi27hi.of.TN._c1, fig.width=12}
fileName <- "results/flowCorrel_pctCD38hi27hi.of.TN..genes.xlsx"
unlink(fileName)
idx <- !is.na(as.numeric(flow.mtx[, "pctCD38hi27hi.of.TN."]))
correlGenes <- clinCorrel(flow.mtx[idx, "pctCD38hi27hi.of.TN."], , combat_edata[rownames(res), flow.patients[idx]])
```
